CI: TestDynamicExporterLifecycle

## CI failure

```
panic: test timed out after 10m0s
running tests:
	TestDynamicExporterLifecycle (10m0s)

goroutine 65 [running]:
testing.(*M).startAlarm.func1()
	/root/sdk/go1.22.3/src/testing/testing.go:2366 +0x385
created by time.goFunc
	/root/sdk/go1.22.3/src/time/sleep.go:177 +0x2d

goroutine 1 [chan receive, 10 minutes]:
testing.(*T).Run(0xc000610d00, {0x19faf0b?, 0x0?}, 0x1ab77d8)
	/root/sdk/go1.22.3/src/testing/testing.go:1750 +0x3ab
testing.runTests.func1(0xc000610d00)
	/root/sdk/go1.22.3/src/testing/testing.go:2161 +0x37
testing.tRunner(0xc000610d00, 0xc000033c60)
	/root/sdk/go1.22.3/src/testing/testing.go:1689 +0xfb
testing.runTests(0xc000602420, {0x2a72220, 0x10, 0x10}, {0x1?, 0x52638e?, 0x2a95d40?})
	/root/sdk/go1.22.3/src/testing/testing.go:2159 +0x445
testing.(*M).Run(0xc00045e280)
	/root/sdk/go1.22.3/src/testing/testing.go:2027 +0x68b
main.main()
	_testmain.go:113 +0x193

goroutine 38 [chan send, 10 minutes]:
github.com/cilium/cilium/pkg/hubble/exporter.(*configWatcher).Stop(0xc00024a1c0?)
	/host/pkg/hubble/exporter/config_watcher.go:81 +0x6d
github.com/cilium/cilium/pkg/hubble/exporter.(*DynamicExporter).Stop(0xc00034b2c0)
	/host/pkg/hubble/exporter/dynamic_exporter.go:58 +0xc5
github.com/cilium/cilium/pkg/hubble/exporter.TestDynamicExporterLifecycle(0xc000348340)
	/host/pkg/hubble/exporter/dynamic_exporter_test.go:40 +0x1ef
testing.tRunner(0xc000348340, 0x1ab77d8)
	/root/sdk/go1.22.3/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	/root/sdk/go1.22.3/src/testing/testing.go:1742 +0x390

goroutine 16 [sync.Mutex.Lock, 10 minutes]:
sync.runtime_SemacquireMutex(0x30220a0?, 0x20?, 0x30220a0?)
	/root/sdk/go1.22.3/src/runtime/sema.go:77 +0x25
sync.(*Mutex).lockSlow(0xc00034b300)
	/root/sdk/go1.22.3/src/sync/mutex.go:171 +0x15d
sync.(*Mutex).Lock(...)
	/root/sdk/go1.22.3/src/sync/mutex.go:90
sync.(*RWMutex).Lock(0x3?)
	/root/sdk/go1.22.3/src/sync/rwmutex.go:146 +0x31
github.com/cilium/cilium/pkg/hubble/exporter.(*DynamicExporter).onConfigReload(0xc00034b2c0, {0x1c8bcc8, 0x30220a0}, 0x31b7b661343ab32f, {{0xc00032e1c0?, 0x16b83a0?, 0xc000718401?}})
	/host/pkg/hubble/exporter/dynamic_exporter.go:85 +0x9d
github.com/cilium/cilium/pkg/hubble/exporter.(*configWatcher).reload(0xc00024a1c0)
	/host/pkg/hubble/exporter/config_watcher.go:72 +0xc3
github.com/cilium/cilium/pkg/hubble/exporter.NewConfigWatcher.func1()
	/host/pkg/hubble/exporter/config_watcher.go:57 +0x33
created by github.com/cilium/cilium/pkg/hubble/exporter.NewConfigWatcher in goroutine 38
	/host/pkg/hubble/exporter/config_watcher.go:51 +0x325
FAIL	github.com/cilium/cilium/pkg/hubble/exporter	600.180s
```

--- a/pkg/hubble/exporter/dynamic_exporter.go
+++ b/pkg/hubble/exporter/dynamic_exporter.go
@@ -52,10 +52,10 @@ func (d *DynamicExporter) OnDecodedEvent(ctx context.Context, event *v1.Event) (
 
 // Stop stops configuration watcher  and all managed flow log exporters.
 func (d *DynamicExporter) Stop() error {
+	d.watcher.Stop()
+
 	d.mutex.Lock()
 	defer d.mutex.Unlock()
-
-	d.watcher.Stop()
 
 	var errs error
 	for _, me := range d.managedExporters {

